src/main/scala/chisel3/util/Decoupled.scala

```diff
@@ -215,9 +217,12 @@ class Queue[T <: Data](val gen: T,
 
   val io = IO(new QueueIO(genType, entries))
 
-  val ram = Mem(entries, genType)
+  val ram = if (useSyncReadMem) SyncReadMem(entries, genType) else Mem(entries, genType)
```

Comment:
```suggestion
  val ram = if (useSyncReadMem) SyncReadMem(entries, genType, SyncReadMem.WriteFirst) else Mem(entries, genType)
```

Agreeing with @sequencer here, it's probably good to specify `WriteFirst` since it's necessary for correctness. While the default undefined R/W-collision behavior maps to write-first in the default flow, this isn't explicitly specified.

Reply:
Once again thanks for suggesting the change. It's included in the latest code.